src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+  const dateParts = date.split(fromSeparator);
+  const dateMap = fromFormat.reduce((map, part, index) => {
+    map[part] = dateParts[index];
+
+    return map;
+  }, {});
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  } else if (dateMap['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? `20${dateMap['YY']}` : `19${dateMap['YY']}`;
```

Comment:
The conversion from 'YY' to 'YYYY' is correct, but similarly to the previous comment, ensure to delete the 'YY' key from `dateMap` after the conversion.